middleware/kubernetes/kubernetes_test.go

```diff
@@ -23,3 +23,119 @@ func TestSymbolContainsWildcard(t *testing.T) {
+	}
+
+	// Test wildcard acceptance
+	query = "*.any.*.any.svc.inter.webs.test."
+	r, e = k.parseRequest(query, "SRV")
+	if e != nil {
+		t.Errorf("Expected no error from parseRequest(%v). Instead got '%v'.", query, e)
+	}
+	if expected := "*"; r.port != expected {
+		t.Errorf("Expected parseRequest(%v) to get port == \"%v\". Instead got '%v'.", query, expected, r.port)
+	}
+	if expected := "any"; r.protocol != expected {
+		t.Errorf("Expected parseRequest(%v) to get protocol == \"%v\". Instead got '%v'.", query, expected, r.protocol)
+	}
+	if expected := ""; r.endpoint != expected {
+		t.Errorf("Expected parseRequest(%v) to get endpoint == \"%v\". Instead got \"%v\".", query, expected, r.endpoint)
+	}
+	if expected := "*"; r.service != expected {
+		t.Errorf("Expected parseRequest(%v) to get service == \"%v\". Instead got \"%v\".", query, expected, r.service)
+	}
+	if expected := "any"; r.namespace != expected {
+		t.Errorf("Expected parseRequest(%v) to get namespace == \"%v\". Instead got '%v'.", query, expected, r.namespace)
+	}
+	if expected := "svc"; r.typeName != expected {
+		t.Errorf("Expected parseRequest(%v) to get typeName == \"%v\". Instead got '%v'.", query, expected, r.typeName)
+	}
+	if expected := "inter.webs.test"; r.zone != expected {
+		t.Errorf("Expected parseRequest(%v) to get zone == \"%v\". Instead got '%v'.", query, expected, r.zone)
+	}
+
+	// Test A request of endpoint
+	query = "1-2-3-4.webs.mynamespace.svc.inter.webs.test."
+	r, e = k.parseRequest(query, "A")
+	if e != nil {
+		t.Errorf("Expected no error from parseRequest(%v). Instead got '%v'.", query, e)
+	}
+	if expected := ""; r.port != expected {
+		t.Errorf("Expected parseRequest(%v) to get port == \"%v\". Instead got \"%v\".", query, expected, r.port)
+	}
+	if expected := ""; r.protocol != expected {
+		t.Errorf("Expected parseRequest(%v) to get protocol == \"%v\". Instead got \"%v\".", query, expected, r.protocol)
+	}
+	if expected := "1-2-3-4"; r.endpoint != expected {
+		t.Errorf("Expected parseRequest(%v) to get endpoint == \"%v\". Instead got \"%v\".", query, expected, r.endpoint)
+	}
+	if expected := "webs"; r.service != expected {
+		t.Errorf("Expected parseRequest(%v) to get service == \"%v\". Instead got \"%v\".", query, expected, r.service)
+	}
+	if expected := "mynamespace"; r.namespace != expected {
+		t.Errorf("Expected parseRequest(%v) to get namespace == \"%v\". Instead got \"%v\".", query, expected, r.namespace)
+	}
+	if expected := "svc"; r.typeName != expected {
+		t.Errorf("Expected parseRequest(%v) to get typeName == \"%v\". Instead got \"%v\".", query, expected, r.typeName)
+	}
+	if expected := "inter.webs.test"; r.zone != expected {
+		t.Errorf("Expected parseRequest(%v) to get zone == \"%v\". Instead got \"%v\".", query, expected, r.zone)
+	}
+
+	// Test invalid A request (with port protocol)
+	query = "_http._tcp.webs.mynamespace.svc.inter.webs.test."
```

Comment:
could you add some malformed requests here? `_http._tcp` and just `.` and `_tcp.test.` maybe?